salt/modules/solarispkg.py

```diff
@@ -342,14 +374,56 @@ def install(name=None, sources=None, saltenv='base', **kwargs):
         return {}
 
     try:
+        # instatiate a state loader so we can use the file.cached state
+        states = salt.loader.states(
+            __opts__,
+            __salt__,
+            __utils__,
+            salt.loader.serializers(__opts__))
         if 'admin_source' in kwargs:
-            adminfile = __salt__['cp.cache_file'](kwargs['admin_source'], saltenv)
+            try:
+                cache_result = states['file.cached'](kwargs['admin_source'],
```

Comment:
You should be able to use `__states__['file.cached']` here instead of going through all the loader stuff